src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -603,8 +604,8 @@ private async Task PromoteTaskToActiveAsync(BackgroundRestoreOperation restoreOp
                 return await _lockService.Value.ExecuteNuGetOperationAsync(async () =>
                 {
                     var componentModel = await _componentModel.GetValueAsync(jobCts.Token);
-
-                    using (var logger = componentModel.GetService<RestoreOperationLogger>())
+                    var outputConsoleProvider = componentModel.GetService<IOutputConsoleProvider>();
```

Comment:
Wouldn't it be nicer to just import the IOutputConsoleProvider through MEF rather than calling the component model?

You can import it lazily. Then you don't need the componentmodel at all.

Reply:
@nkolev92  It looks if I use [Import] inside RestoreOperationLogger then I have re-introduce [ImportingConstructor] or create another component model to make MEF importing working.

Reply:
I think Nikolche means import though MEF in this class. It already has an `[ImportingConstructor]`.

Maybe `Lazy<ISolutionRestoreJob>` could also be imported through the constructor, and then we wouldn't need either `componentModel` or `_componentModel`.

Reply:
Yeah, I meant to import the IOutputConsoleProvider in this class so you only do it once. 

ISolutionRestoreJob is once per operation, you can't really import it in the solutionrestoreworker.

Reply:
@nkolev92 @zivkan Yes, MEF import for IOutputConsoleProvider, ISolutionRestoreJob works.
But only thing is it keeps below thing from `RestoreOperationLogger` in memory even after solution is closed. But with current implementation I don't see anything left from `RestoreOperationLogger` in memory. Please advice if I go ahead with MEF import for IOutputConsoleProvider, ISolutionRestoreJob.

![image](https://user-images.githubusercontent.com/8766776/84540701-ce132380-acaa-11ea-8ebf-d57754fb232b.png)

Reply:
I'm not sure how changing how IOutputConsoleProvider is acquired can affect the RestoreOperationLogger being in memory. 
The restore operation logger should be used the same way in either situation. 

Does RestoreOperationLogger subscribe to something from IOutputConsoleProvider that doesn't get cleared when the logger is disposed?

Reply:
@nkolev92  I made IOutputConsoleProvider lazy initializer. But can't do same thing with `ISolutionRestoreJob`, because it's has `NotShared` attribute, that means MEF have to create new instance for everytime. 
If I inject it through [ImportingConstructor] for SolutionRestoreWorker we have problem, because SolutionRestoreWorker is singleton `Shared`. So `ISolutionRestoreJob` would be created only once together with SolutionRestoreWorker, but another new instance is not available. It looks only way create another instance besides constructor in Singletone is yet another MEF container.
I guess this what we can get for now.

Reply:
This last message doesn't really match the code anymore. 

I actually agree with the code and that was my recommendation :)